### PR TITLE
Add Post Comment block

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Related: https://github.com/WordPress/gutenberg/issues/24101
Laying the foundation for a block-based post comment building. Adds a **Post Comment** block which requires a Comment ID input for now. The **Post Comment** block provides a context for the inner blocks.

As an example, also implemented a very simple **Post Comment Content** block, which uses the context provided by the **Post Comment** block.

## How has this been tested?
1. Open site editor
2. Add **Post Comment** block
3. Input an existing comment's ID
4. Insert **Post Comment Content** block into the **Post Comment** block
5. You should see the content of the comment

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/post-comment-content/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
```
Comment (ockham) [inline packages/block-library/src/post-comment-content/index.js:5]: Let's maybe

```suggestion
import { comment as icon } from '@wordpress/icons';

```

here, and use it as `icon` further down.

File: packages/block-library/src/post-comment-content/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import metadata from './block.json';
+import edit from './edit';
+
+const { name } = metadata;
+export { metadata, name };
+
+export const settings = {
+	title: __( 'Post Comment Content' ),
+	description: __( 'Post Comment Content' ),
+	edit,
```
Comment (ockham) [inline packages/block-library/src/post-comment-content/index.js:18]:
```suggestion
	edit,
	icon,
```

**Review by ockham — CHANGES_REQUESTED** [review]
Unit tests are failing because of missing fixtures for the two new blocks that this PR introduces. [This doc](https://github.com/WordPress/gutenberg/blob/710373b254fbcd15d524afdeb31da0d93c4defd4/packages/e2e-tests/fixtures/blocks/README.md#creating-fixtures) has some information on how to create them :slightly_smiling_face:

**Comment by david-szabo97** [comment]
@ockham Gotcha! Added the icons and fixtures. Tests are happy now!

File: packages/block-library/src/post-comment-content/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { alignJustify as icon } from '@wordpress/icons';
```
Comment (ockham) [inline packages/block-library/src/post-comment-content/index.js:5]: Sorry for the nitpick, but let's not use `alignJustify` as icon :sweat_smile: It's just a bit too confusing to see any of the alignment-related icons as the block type icon (it's a really unexpected place).

If you don't wanna use `comment`, we can look for something else that represents written content :thinking:

Comment (david-szabo97) [reply packages/block-library/src/post-comment-content/index.js:5]: Actually I stole it from Post Content 😅 `comment` works fine if you think `alignJustify` is inappropriate here.

https://github.com/WordPress/gutenberg/blob/beced056e067f5cc34919a31b216491da267a7cf/packages/block-library/src/post-content/index.js#L5

Comment (ockham) [reply packages/block-library/src/post-comment-content/index.js:5]: > Actually I stole it from Post Content :sweat_smile: 

Oh, I didn't realize! :sweat_smile: 

Okay, while that's not optimal, we can land as-is and defer to designers to get better icons for this.

**Review by ockham — APPROVED** [review]
Looking good! This will provide us with a good baseline to work on other comment related blocks.

**Pull request merged**